tools/2023-11-interactive-commands-functional-tests.md

```diff
@@ -0,0 +1,124 @@
+
+## Design
+
+* <https://github.com/Netflix/go-expect>: "An expect-like golang library to automate control of terminal or console based programs."
+* <https://github.com/hinshun/vt10x>: "About Package vt10x is a vt10x terminal emulation backend."
+* <https://github.com/creack/pty>: "PTY (PseudoTeletYpe) interface for Go."
+
+### Design details
+
+<!--
+This section should be detailed and thorough enough that another developer
+could implement your design and provide enough detail to get a high confidence
+estimate of the cost to implement the feature but isn’t as detailed as the 
+code. Be sure to also consider testability in your design.
+
+For each change, give each "change" in the proposal its own section and
+describe it in enough detail that someone else could implement it. Cover
+ALL of the important decisions like names. Your goal is to get an agreement
+to proceed with coding and PRs.
+
+If there are alternatives you are considering please include that in the open
+questions section. If the product has a layered architecture, it's good to
+align these sections with the product's layers. This will help readers use
+their current understanding to understand your ideas.
+
+* Advantages of this design - Describe what's good about this plan relative to
+  other options. Does it feel easy to implement? Provide flexibility for
+  future work?
+* Disadvantages - Describe what's not ideal about this plan. If you don't
+  point these things out other people will do it for you. This is a good place
+  to cover risks.
+-->
+
+### API design (if applicable)
+
+* New structs to be added: Prompt, and InteractiveCommandRunner.
+* A Prompt will have Command Arguments array and Procedure which is an array of interactive actions. Procedure will mostly consist of expectations with the help of Netflix's `go-expect` library.
+* An InteractiveCommandRunner will have Run method which will take a Prompt.
+
+## Alternatives considered
+
+<!--
+Describe the alternative designs that were considered or should be considered.
+Give a justification for why alternative approaches should be rejected if
+possible. 
+-->
+
+## Test plan
+
+<!--
+Include the test plan to validate the features including the areas that
+need functional tests.
+
+Describe any functionality that will create new testing challenges:
+- New dependencies
+- External assets that tests need to access
+- Features that do I/O or change OS state and are thus hard to unit test
+-->
+
+## Security
+
+* Some tests will need access to the account ids and/or secrets of, for example, AWS and Azure. At that moment, we are going to get these variables/secrets from the repository context which means that there will be no addition of these variables to the code therefore to the Radius repository.
```

Comment:
To run the test, does user has to explicitly add these variables after the fork?